### PR TITLE
Updated for Bannerlord 1.7.2

### PR DESCRIPTION
**Pull request by OrderWOPower**
I updated Party Screen Enhancements for Bannerlord 1.7.2. Enjoy!

### CODE REVIEW

File: PartyScreenEnhancements/ModuleData/SubModule.xml [hunk 1]
```diff
@@ -3,15 +3,15 @@
...
+	<Version value="v1.6.5" />
 	<SingleplayerModule value="true" />
 	<MultiplayerModule value="false" />
 	<DependedModules>
-		<DependedModule Id="Native" DependentVersion="e1.7.0"/>
-		<DependedModule Id="SandBoxCore" DependentVersion="e1.7.0"/>
-		<DependedModule Id="Sandbox" DependentVersion="e1.7.0"/>
-		<DependedModule Id="CustomBattle" DependentVersion="e1.7.0"/>
-		<DependedModule Id="StoryMode" DependentVersion="e1.7.0"/>
+		<DependedModule Id="Native" DependentVersion="e1.6.5"/>
```
Comment (Hirtol) [inline PartyScreenEnhancements/ModuleData/SubModule.xml]: These version numbers should reference 1.7.2, right?

File: PartyScreenEnhancements/SubModule.cs [hunk 2]
```diff
@@ -26,12 +26,12 @@ protected override void OnSubModuleLoad()
                 FileLog.Log($"PSE Config Load Exception: {e}");
             }
 
-            //UIResourceManager.UIResourceDepot.StartWatchingChangesInDepot();
+            UIResourceManager.UIResourceDepot.StartWatchingChangesInDepot();
```
Comment (Hirtol) [inline PartyScreenEnhancements/SubModule.cs]: This should probably be disabled for the Nexus build

File: PartyScreenEnhancements/ViewModel/Settings/Tabs/Sorting/SettingSortVM.cs [hunk 3]
```diff
@@ -22,23 +22,23 @@ public class SettingSortVM : TaleWorlds.Library.ViewModel
 
         public SettingSortVM(PartySort sortingComparer, Action<SettingSortVM, SettingSide> transferCallBack, SettingSide side, Action<SettingSortVM> openSubSetting)
         {
-            this.SortingComparer = sortingComparer;
-            this.SettingHint = new HintViewModel(new TextObject(SortingComparer.GetHintText()));;
-            this._transferHint = new HintViewModel(new TextObject($"Click to transfer to the {side.GetOtherSide().ToString().ToLower()} side!"));;
-            this.AscDescHint = new HintViewModel(new TextObject($"Current Mode: {(IsDescending ? "Descending" : "Ascending")}"));;
+            SortingComparer = sortingComparer;
+            SettingHint = new HintViewModel(new TextObject(SortingComparer.GetHintText())); ;
```
Comment (Hirtol) [inline PartyScreenEnhancements/ViewModel/Settings/Tabs/Sorting/SettingSortVM.cs]: Double semicolons

File: PartyScreenEnhancements/ViewModel/SortAllTroopsVM.cs [hunk 4]
```diff
@@ -108,20 +101,13 @@ public void SortTroops()
                 toSort.Insert(0, leaderVm);
             }
 
-            // Here we manually clear the roster while ignoring the party leader.
-            // Don't use `rosterToSort.Clear()` as that seems to cause the party leader to get unset permanently afterward, which stops upgrades from working.
-            rosterToSort.RemoveIf((item) => item.Character != leaderOfParty);
+            rosterToSort.Clear();
```
Comment (Hirtol) [inline PartyScreenEnhancements/ViewModel/SortAllTroopsVM.cs]: Is the above comment no longer relevant? Might be useful to leave a comment why clearing the roster is now safe

File: PartyScreenEnhancements/ViewModel/SortAllTroopsVM.cs [hunk 5]
```diff
@@ -87,19 +86,13 @@ public void SortTroops()
...
-            TroopRoster rosterToSort, PartySort sorter)
+        private static void SortAnyParty(MBBindingList<PartyCharacterVM> toSort, PartyBase party, TroopRoster rosterToSort, PartySort sorter)
         {
             if (rosterToSort == null || rosterToSort.Count == 0 || toSort == null || toSort.IsEmpty()) return;
 
-            CharacterObject leaderOfParty = party?.LeaderHero?.CharacterObject;
-
-            // Sort the list, this is done for the visual unit cards to be properly positioned after the sort
-            // This is not yet persisted to the actual roster, that is done after this.
+            var leaderOfParty = party?.LeaderHero.CharacterObject;
```
Comment (Hirtol) [inline PartyScreenEnhancements/ViewModel/SortAllTroopsVM.cs]: I'd prefer to keep the comments explaining the details of why we're doing the following

File: PartyScreenEnhancements/ViewModel/SortAllTroopsVM.cs [hunk 6]
```diff
@@ -108,20 +101,13 @@ public void SortTroops()
                 toSort.Insert(0, leaderVm);
             }
 
-            // Here we manually clear the roster while ignoring the party leader.
-            // Don't use `rosterToSort.Clear()` as that seems to cause the party leader to get unset permanently afterward, which stops upgrades from working.
-            rosterToSort.RemoveIf((item) => item.Character != leaderOfParty);
+            rosterToSort.Clear();
 
-            // Re-add the correctly sorted troops to the roster. We need to do it in this janky way due to the fact that we can't easily sort
```
Comment (Hirtol) [inline PartyScreenEnhancements/ViewModel/SortAllTroopsVM.cs]: Probably keep this comment as well

**Review by Hirtol — CHANGES_REQUESTED** [review]
Thank you for taking the time to go through this!

The main question for me revolves around the `rosterToSort.Clear();` call. It used to corrupt people's saves so I'm a bit hesitant about it.

It seems Taleworlds added new Dlls which I forgot to import, which broke the ScreenLayer hack. Ended up giving up on updating it a while back due to this breaking. Something to keep in mind for myself in the future.

**Comment by OrderWOPower** [comment]
Hmm, I might have forked a wrong branch... Will try again later.

**Review by Hirtol — COMMENTED** [review]
Thanks again, I'll merge this and upload it to the Nexus.

**Review by Hirtol — APPROVED** [review]

**Pull request merged**